python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,43 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self):
+        node_ips = [
+            self.provider.internal_ip(node_id)
+            for node_id in self.nodes_to_terminate
+        ]
+        raylet_ids_to_drain = {
+            self.load_metrics.raylet_id_by_ip[ip]
+            for ip in node_ips
+        }
+
+        logger.info(f"Draining raylets with ids {raylet_ids_to_drain}.")
+        request = gcs_service_pb2.DrainNodeRequest(drain_node_data=[
+            gcs_service_pb2.DrainNodeData(node_id=raylet_id)
+            for raylet_id in raylet_ids_to_drain
+        ])
+        try:
+            response = self.gcs_node_info_stub.DrainNode(request)
+            drained_raylet_ids = {
+                status_item.node_id
+                for status_item in response.drain_node_status
+            }
+            failed_to_drain = raylet_ids_to_drain - drained_raylet_ids
+            if failed_to_drain:
+                logger.error(
+                    f"Failed to drain raylets with ids {failed_to_drain}.")
+        except grpc._channel._InactiveRpcError as e:
```

Comment:
won't we get that if the operator is down or if there is a network issue? i definitely saw these errors happen unrelated to ray 1.8

Reply:
In the next line, we check if the RPC error stems specifically from the method being unimplemented on the server.
If not, we raise the error. 

Will add some comments to make this clearer.

Reply:
Comments added.